File: mi_conjunto_guardas/lib/models/residential_block.dart


/// Modelo de 'conjunto_residencial'
class ResidentialBlock {
  final int residentialBlockId;
  final String email;
  final String name;

  ResidentialBlock({this.residentialBlockId, this.email, this.name});

  factory ResidentialBlock.fromJson(Map<String, dynamic> json) {
    return ResidentialBlock(
        residentialBlockId: json["residentialBlockId"],
        email: json["email"],
        name: json["name"]);
  }
}

File: server/sequelize/models/ResidentialBlock.js
import DataTypes from "sequelize";

/**
 * Modelo de 'conjunto_residencial' en base de datos
 * @param {*} sequelize 
 */
export const residentialBlockModel = (sequelize) => {
  sequelize.define(
    "ResidentialBlock",
    {
      residentialBlockId: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
        field: "id_conjunto_residencial",
      },
      name: {
        type: DataTypes.STRING,
        allowNull: false,
        field: "nombre",
      },
      email: {
        type: DataTypes.STRING,
        allowNull: false,
        unique: true,
      },
      password: {
        type: DataTypes.STRING,
        allowNull: false,
      },
    },
    {
      tableName: "conjunto_residencial",
      timestamps: false,
    }
  );
};
